cmd/create/machinepool/nodepool.go

```diff
@@ -31,9 +32,29 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		os.Exit(1)
 	}
 
-	// Hosted clusters create identifiers for NodePools, users don't interact directly with these resources
-	if cmd.Flags().Changed("name") {
-		r.Reporter.Errorf("Setting the `name` is not supported for hosted clusters")
+	// Machine pool name:
+	name := strings.Trim(args.name, " \t")
```

Comment:
We have a `hypershift.CleanName` already, maybe that function can be updated to replace `\t` as well and reused across nodepools.

Reply:
Ah, the function is in the CS code. But might be worth looking into how to works so that we're consistent between API and CLI

Reply:
Note that I'm mimic-ing here the same behaviour done in the classic rosa for machinepool.
I.e. https://github.com/openshift/rosa/blob/master/cmd/create/machinepool/machinepool.go#L59

I mean, I tried to be consistent both and this logic should be extracted for both cases (and changed in both cases if it's not correct).

Reply:
AFAIK this becomes the name of a NodePool resource no? So it means it must be a valid kube resource name (for example, no underscore)

Reply:
It's part of the name in the kube resource.
There is already a regex that checks if the name is correct.

I think it's good for kube (unless I missed something)

```
// Regular expression to used to make sure that the identifier given by the
// user is safe and that it there is no risk of SQL injection:
var machinePoolKeyRE = regexp.MustCompile(`^[a-z]([-a-z0-9]*[a-z0-9])?$`)
```